### ISSUE TITLE
ppx attributes on polymorphic variants get pretty printed away

### ISSUE THREAD

**Issue by Schmavery** [facebook/reason#821]
We (@bsansouci @sgrove) were playing with interfacing with webGL yesterday and ran into this.

```
external foo : something => ([`array_buffer [@bs.as 12353] | `array_something_buffer [@bs.as 1231232132]]) => unit = "foo" [@@bs.send];
```

pretty prints to

```
external foo : something => ([`array_buffer | `array_something_buffer ]) => unit = "foo" [@@bs.send];
```

**Comment by bsansouci** [comment]
@bobzhang ^

**Comment by sgrove** [comment]
Hitting this with the BuckleScript example:

```
external readFileSync :
  name:string ->
  ([ `utf8
   | `my_name [@bs.as "ascii"]
   ] [@bs.string]) ->
  string = ""
  [@@bs.module "fs"]
```
 
Goes to:

```
external readFileSync : name::string => [ | `utf8 | `my_name] [@bs.string] => string =
  "" [@@bs.module "fs"];
```

**Comment by IwanKaramazow** [comment]
I'll fix it today.

**Comment by IwanKaramazow** [comment]
@sgrove PR is awaiting review 🏊

**Comment by wokalski** [comment]
This one can be closed (resolved by #1109).
cc @chenglou

**Comment by chenglou** [comment]
Thanks!
